### PR TITLE
Feat/company create update

### PR DESCRIPTION
**Pull request by Parle648**
## Describe your changes

- Create Create|Update|Get all endpoints for company & apply roles guard.

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://codecraftersintership.atlassian.net/browse/SCRUM-12?atlOrigin=eyJpIjoiZDNkY2UzOWI4ZTQ3NGQ0ZGI5MGIwYmQ0MDRjNjViNWQiLCJwIjoiaiJ9)

### **General**

- [x] types for input and output params
- [x] no `any`, should be also added to eslint rules
- [x] try/catch for any async function
- [x] no **magic** [numbers](/7d77574ee4bc4e339a68066762e887cb)
- [x] compare only with constants not with strings (instead of user === ‘admin’, better user === roles.admin)
- [x] no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] avoid similar types with duplicating by generics
- [x] no **console.log** in pr
- [x] .env file should be in .gitignore
- [x] delete commented code if it's not part of documentation
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file
- [x] **constants** and **function** names should be lowercase.
- [x] no dates format in the code (like "dd/MM/yyyy”), move it in global constant variable
- [x] import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] strings should be in the constant variable. Example: instead of ‘15 3 \* \* _’, const EACH_DAY_15h_15min = ‘15 3 _ \* \*’

### Backend

- [x] swagger for each endpoint. add in the documentation, different types of responses.
      on a successful response (2xx), on unsuccessful response (4xx, 5xx)
- [x] less requests to db, all data should be taken with one query
- [x] **public** in method use only is use function externally
- [x] use ConfigService instead of **process.env**
- [x] use @`@index` decorator for frequently requested data
- [x] use transactions if there is a call chain that mutates data in different tables
- [x] add a set of rules for [nestjs](https://github.com/darraghoriordan/eslint-plugin-nestjs-typed) to the .eslint config file. It helps to prevent bugs and confusing moments
- [x] use [REST API Naming Convention](https://medium.com/@nadinCodeHat/rest-api-naming-conventions-and-best-practices-1c4e781eb6a5) for endpoints

  P.S: more information about RESTful API design by [Microsof](https://learn.microsoft.com/en-us/azure/architecture/best-practices/api-design)t

- [x] use ‘**UUIDs**` for primary key, ** to prevent Enumeration Attacks**

### CODE REVIEW

File: src/modules/company/company.controller.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+@Controller('company')
+export class CompanyController {
+  constructor(private readonly companyService: CompanyService) {}
+
+  @Get()
+  @UseGuards(RolesGuard)
+  @SetMetadata('roles', [Roles.SUPERADMIN])
+  @ApiOperation({ summary: 'Get list of companies' })
+  @ApiResponse({ status: 200, type: [Company] })
+  async getAll() {
```
Comment (lyudad) [inline src/modules/company/company.controller.ts]: types

File: src/modules/company/company.service.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      return await this.entityManager.save(company);
+    } catch (error) {
+      throw new InternalServerErrorException('Internal server error');
+    }
+  }
+
+  async getList(): Promise<Company[]> {
+    try {
+      const companies = await this.companyRepo.find();
+      return await this.entityManager.save(companies);
```
Comment (lyudad) [inline src/modules/company/company.service.ts]: why this needed, you just need return companies list

File: src/modules/company/company.service.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    } catch (error) {
+      throw new InternalServerErrorException('Internal server error');
+    }
+  }
+
+  async update(id: number, updateCompanyDto: UpdateCompanyDto): Promise<{ status: number; message: string }> {
+    try {
+      const company: UpdateResult = await this.companyRepo.update(+id, updateCompanyDto);
```
Comment (lyudad) [inline src/modules/company/company.service.ts]: why + needed, if is is number in types

Comment (Parle648) [reply src/modules/company/company.service.ts]: remove + operator

File: src/modules/company/company.service.ts (hunk 2 above)
Comment (Parle648) [reply src/modules/company/company.service.ts]: delete useless entityManager call

File: src/modules/company/company.controller.ts (hunk 1 above)
Comment (Parle648) [reply src/modules/company/company.controller.ts]: added types

File: src/modules/company/company.service.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  async getList(): Promise<Company[]> {
+    try {
+      const companies: Company[] = await this.companyRepo.find();
+      return companies;
+    } catch (error) {
+      throw new InternalServerErrorException('Internal server error');
+    }
+  }
+
+  async update(id: number, updateCompanyDto: UpdateCompanyDto): Promise<{ status: number; message: string }> {
```
Comment (lyudad) [inline src/modules/company/company.service.ts]: { status: number; message: string } - you have this type couple times, move it to reusable Interface

Comment (Parle648) [reply src/modules/company/company.service.ts]: ok

**Pull request merged**